packages/resource-deployment/scripts/function-app-create.sh

```diff
@@ -32,23 +34,30 @@ if [ -z "$resourceGroupName" ]; then
     exitWithUsageInfo
 fi
 
+appRegistrationName="allyappregistration"
+
+# create the app registration if it doesn't exist
+if ! az ad app show --id $clientId 1>/dev/null; then
```

Comment:
either use create --id to pass the client id or use display name to check if the id exists using the az command az ad app list --display-name "<principal_name>"

Reply:
i prefer display name option

Reply:
How?, id is required for "az ad app show", we can't use the name, and even if "az ad app show" support searching only by name,  it might not be unique, you can have two registration app with the same name